src/CountableListInterface.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+namespace Dhii\Collection;
+
+use Traversable;
+use Countable;
+
+/**
+ * A list that can be counted.
+ *
+ * @since [*next-version*]
+ */
+interface CountableListInterface extends
```

Comment:
From the looks of it, this is the "base" interface in this package. But I can't see how or why anything would require traversing _and_ counting. I was expecting `Traversable` to be the base interface for everything here, tbh.